routers/api/v1/repo/hook.go

```diff
@@ -96,6 +96,19 @@ func CreateHook(ctx *context.APIContext, form api.CreateHookOption) {
 	ctx.JSON(201, convert.ToHook(ctx.Repo.RepoLink, w))
 }
 
+func DeleteHook(ctx *context.APIContext) {
+	if err := models.DeleteWebhook(ctx.ParamsInt64(":id")); err != nil {
+		if models.IsErrWebhookNotExist(err) {
```

Comment:
There is no possibility to return this type of error, you only need to response 500 when error occurs.